microsoftgraph/microsoft-graph-explorer-v4
# Fix: use resources to validate paths

## Overview
Walking through the resources ensures that we have a working model for the validator to go through. This guarantees that we are working off a known set of resources . 
All queries that do not match a known path created by combining segments in the resource tree are marked as invalid which we can validate with certainity

Fixes #2461

### Demo

Optional. Screenshots, `curl` examples, etc.

### Notes

[The ABNF rules in the ODATA ABNF repository are documentation intended for human readers. They are not intended for generating a ready-to-use URL parser..](https://github.com/oasis-tcs/odata-abnf/issues/99#issuecomment-1600323468)

## Testing Instructions

* Spin up GE
* In the Query Bar, write anything after the /v1.0/ 
* Notice the messaging about it not being a valid URL
